### ISSUE TITLE
Select question toggle pop up displayed at the top of page instead of by the problem

### ISSUE THREAD

**Issue by bhoffman0** [RunestoneInteractive/RunestoneComponents#1226]
When you choose a question in selectquestion:: :toggle: exercises, the pop up appears at the top of the page instead of by the question, and you don't see it unless you scroll up. Can it be put by the question?
![image](https://user-images.githubusercontent.com/4571962/126876012-91c8286c-9de6-4f80-b7aa-7e6011312bee.png)

**Comment by bnmnetp** [comment]
This is how they appear in the overview book:
<img width="1193" alt="Screen Shot 2021-07-24 at 1 02 48 PM" src="https://user-images.githubusercontent.com/51115/126877362-67b2d7da-1ec5-49eb-a0a3-f721eae7f658.png">

Are you viewing them in an assignment on the doAssignment page?

**Comment by bhoffman0** [comment]
No, this is from a page in the book: https://csawesome.runestone.academy/runestone/books/published/csawesome/Unit1-Getting-Started/topic-1-8-practice-mixed-code-toggle.html
Try to toggle the second exercise and you have to scroll up to see the pop up. I tried it on a page in Barb's py4e book too and it did the same.

**Comment by bnmnetp** [comment]
Ah -- 

@vqum -- We should make it so that each toggle question has its own unique id for the toggle preview.  Otherwise students may click to preview the question and then have to scroll way up the page to see the preview.

I think the fix is as easy as adding the id for the toggle question to the id  so that it is `<div id="xxxx-toggle-preview">

We should prefer toggle-preview over component-preview on the assignment page as well.

Looking at the code I can see you designed and wrote it to act the way it is now so maybe you and @barbarer had something in mind that @bhoffman0 and I are missing.

**Comment by vqum** [comment]
Barb noticed this earlier and mentioned it to me as well. So far our hypothesis is that specifically the questions built into textbooks have this issue with the select toggle preview pane. Here's another example from Python for Everyone:![image](https://user-images.githubusercontent.com/40499850/126880314-bdd32194-0553-466d-a2e7-bee7f0fbbfb0.png)![image](https://user-images.githubusercontent.com/40499850/126880300-5cf33e70-71ac-4f94-9fea-49beee87722b.png)

However, questions built into assignment pages do not have this issue:
![image](https://user-images.githubusercontent.com/40499850/126880314-bdd32194-0553-466d-a2e7-bee7f0fbbfb0.png)

The difference that I noticed is that on assignment pages, the page CSS includes `#component-preview` with `position: fixed` that allows the preview pane to float over the rest of the page contents while scrolling. The textbook pages CSS does not include `#component-preview` and therefore does not have `position: fixed`

**Comment by bnmnetp** [comment]
If that is what is desired, then it should be as easy as copying the CSS for #component-preview into the selectquestion.css

That would always position the toggle preview on the current page in the upper right.  

Seems like an easy solution to me if @barbarer and @bhoffman0 agree.

**Comment by barbarer** [comment]
Yes, I would like it to appear to the right at the top of the current
view (not just the top of the page).

Dr. Barbara Ericson
Assistant Professor, School of Information
University of Michigan

**Comment by vqum** [comment]
I am also wondering whether scrolling around the page could result in losing track of which question the preview problem corresponds to. @barbarer have you seen that happening at all? Perhaps some sort of label or indicator or a "refocus" button could be added if necessary.

**Comment by barbarer** [comment]
Maybe.  It might be nice if you could scroll some to see more of the
question if the pop-up is covering the question.

Dr. Barbara Ericson
Assistant Professor, School of Information
University of Michigan
